File: 26.删除有序数组中的重复项.cs
/*
 * @lc app=leetcode.cn id=26 lang=csharp
 *
 * [26] 删除有序数组中的重复项
 */

// @lc code=start
public partial class Solution
{
    public int RemoveDuplicates(int[] nums)
    {
        int removeCount = 0;
        int last = int.MaxValue;
        for (int currentIndex = 0; currentIndex < nums.Length; currentIndex++)
        {
            int num = nums[currentIndex];
            if (num == last)
            {
                removeCount++;
            }
            else
            {
                int insertIndex = currentIndex - removeCount;
                if (insertIndex != currentIndex)
                {
                    nums[insertIndex] = num;
                }
            }
            last = num;
        }
        return nums.Length - removeCount;
    }
}
// @lc code=end

File: src/26.删除有序数组中的重复项.rs

/*
 * @lc app=leetcode.cn id=26 lang=rust
 *
 * [26] 删除有序数组中的重复项
 */

// @lc code=start
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut remove_count = 0;
        let mut last = i32::MAX;
        let mut current_index = 0;
        let len = nums.len();
        while current_index < len {
            let current_num = nums[current_index];
            if current_num == last {
                remove_count += 1;
            } else {
                let insert_index = current_index - remove_count;
                nums[insert_index] = current_num;
            }
            last = current_num;
            current_index += 1;
        }
        return (nums.len() - remove_count) as i32;
    }
}
// @lc code=end
